pkg/challenges/challenges.go

```diff
@@ -243,35 +202,6 @@ func kubernetes(ctx context.Context, principal *oidc.IDToken) (identity.Principa
 	}, nil
 }
 
-func githubWorkflow(ctx context.Context, principal *oidc.IDToken) (identity.Principal, error) {
-	workflowRef, err := workflowFromIDToken(principal)
-	if err != nil {
-		return nil, err
-	}
-	additionalInfo, err := workflowInfoFromIDToken(principal)
-	if err != nil {
-		return nil, err
-	}
-
-	cfg, ok := config.FromContext(ctx).GetIssuer(principal.Issuer)
-	if !ok {
-		return nil, errors.New("invalid configuration for OIDC ID Token issuer")
-	}
-
-	issuer, err := oauthflow.IssuerFromIDToken(principal, cfg.IssuerClaim)
```

Comment:
Sorry, just saw this - Don't we still need this? `IssuerFromIDToken` pulls the issuer either from the issuer field or from a field that's specified in the configuration?

Actually, I think the bug might be here - https://github.com/sigstore/fulcio/blob/e1fe2f85ba4dc3277f1c4585e9d3049f72246ee3/pkg/challenges/challenges.go#L413
If you want to frontload fetching the correct issuer, you'll need to do that in `PrincipalFromIDToken`

Reply:
Created https://github.com/sigstore/fulcio/pull/605 to test `IssuerClaim`

Reply:
Ah yes I didn't noticed I'd make a change to the business logic here. I've done the same change in SPIFFE PR and justified its removal here https://github.com/sigstore/fulcio/pull/604#issuecomment-1134998659.

The change is ignoring `FulcioConfig.IssuerClaim` for github actions workflows. This is likely for the best as I can only see this claim being abused / misconfigured for a Github workflow. The claim is only helpful in the case of a federating identity provider like Dex for which the issuer is the federated identity provider (e.g `oauth2.sigstore.dev`) and we'd prefer to place the upstream provider (e.g `https://accounts.google.com`) in the certificate. 

I don't think we should support this form of federating identity provider for github workflow principals. We can add some logic in the config parsing to panic should such an invalid configuration is provided.

If you wish I can break up this change in business logic from the refactor

Reply:
Makes sense - I think this is something we would only support if we either control the OIDC provider or if we have a list of configured federated issuers too, but the latter isn't something we need now.

+1, can you split this off into its own PR? Makes it easier if we need to rollback or if we want to look at the history of changes.

Reply:
Ok sounds good. I'll front-load a change to drop this from Github workflows, SPIFFE and Kubernetes type principals. Should it be dropped from username and URI principals? I don't really understand how those ones are used or will be used right now

Reply:
Yea, let's drop it from those too, since there's not an immediate need.

Reply:
Opened #606. Will rebase on that once merged